mathext/hyp2f1.go

```diff
@@ -0,0 +1,466 @@
+package mathext
+
+import (
+	"fmt"
+	"math"
+)
+
+// Machine epsilon used in Cephes.
+var mACHEP float64 = (math.Nextafter(1.0, 2.0) - 1.0) / 2
+
+const eps float64 = 1e-13
+const ethresh float64 = 1e-12
+
+// Hyp2f1 returns the value of the Gaussian Hypergeometric function at z.
+// For |z| < 1, this implementation follows the Cephes library.
+// For |z| > 1, this implementation perform analytic continuation via relevant Hypergeometric identities.
+// See https://en.wikipedia.org/wiki/Hypergeometric_function for more details.
+func Hyp2f1(a float64, b float64, c float64, z float64) (float64, error) {
+	if math.Abs(z) < 1 {
+		return hyp2f1(a, b, c, z)
+	}
+
+	// Function undefined between the 1 and Inf branch points.
+	if z > 0 {
+		return math.NaN(), fmt.Errorf("analytic continuation failed")
```

Comment:
If we are using errors, please use `errors.New` and if the sentinels are important (this is an open question), export them as package vars rather than forcing client code to do string comparison if they care.